src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -337,7 +341,24 @@ internal static string GetRuntimeClassForTypeCreation(IInspectable inspectable,
 
         private static bool ShouldProvideIReference(Type type)
         {
-            return type.IsValueType || type == typeof(string) || type == typeof(Type) || type.IsDelegate();
+            static bool IsWindowsRuntimeType(Type type)
+            {
+                // TODO: attribute checks are expensive - cache per-vtable
```

Comment:
Is this ToDo still necessary given the higher level caching changes where this function gets called?

Reply:
no, should be covered by that